### PR TITLE
[Ray] Support client

### PR DESCRIPTION
**Pull request by amogkam**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs?
- [x] Did you write any tests to validate this change?  
- [x] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Placement groups do not work with Ray Client in Ray 1.3.

This PR introduces a layer that transparently allows Horovod to work with Ray Client.

We will likely remove this layer in Ray 1.4.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

File: test/single/test_ray.py [hunk 1]
```diff
@@ -372,6 +387,24 @@ def simple_fn(worker):
     hjob.shutdown()
 
 
+def test_remote_client_train(ray_start_client):
```
Comment (richardliaw) [inline test/single/test_ray.py:390]:
```suggestion
@pytest.mark.skipif(
    not gloo_built(), reason='Gloo is required for Ray integration')
def test_remote_client_train(ray_start_client):
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 - Added pytorch_lightning spark estimator which enables training pytorch_lightning models. ([#2713](https://github.com/horovod/horovod/pull/2713))
 
+- Supports Ray Client without code changes. ([#2882](https://github.com/horovod/horovod/pull/2882))
```
Comment (richardliaw) [inline CHANGELOG.md:16]:
```suggestion
- Supports Ray Client without code changes. ([#2882](https://github.com/horovod/horovod/pull/2882))

```

**Review by richardliaw — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Comment by tgaddair** [comment]
@amogkam @richardliaw RE the changes for 1.4, do we still need this layer?

**Comment by richardliaw** [comment]
Nope!

**Pull request merged**